include/s3select_functions.h

```diff
@@ -1190,6 +1192,66 @@ struct _fn_nullif : public base_function {
       }
     };
 
+struct _fn_trim : public base_function {
+
+    std::string buff;
+    std::string buff1;
+    value v_str;
+    value v_next;
+
+    void removespace(std::string& s)
+    {
+      s.erase(remove(s.begin(), s.end(), ' '), s.end());
+    }
+
+    void removespacetrail(std::string& s)
+    {
+      s.erase(std::find_if(s.rbegin(), s.rend(), std::bind1st(std::not_equal_to<char>(), ' ')).base(), s.end());
+    }
+
+    void removespacelead(std::string& s)
+    {
+      s.erase(s.begin(), std::find_if(s.begin(), s.end(), std::bind1st(std::not_equal_to<char>(), ' ')));
+    }
+
+    bool operator()(bs_stmt_vec_t* args, variable* result)
+    {
+      std::string s1("trailing");
+      std::string s2("leading");
+      std::string s3("both");
+      bs_stmt_vec_t::iterator iter = args->begin();
```

Comment:
current interface is temporary,so i suggest to use enum , and set it according to args[0].
and use a switch-case.